### ISSUE TITLE
Getting twin on reconnect always fails

### ISSUE THREAD

**Issue by wmmihaa** [Azure/azure-iot-sdk-node#28]
# Context
- **OS:** Windows 10, Debian
- **Node.js version:** 7.7.2 and 6.5.2
- **npm version:** 4.1.2
- **list of installed packages:** azure-iot-device azure-iot-device-mqtt

# Description of the issue:
Is there a pattern for reconnecting the device client (and twin). In AWS this is done within the client and nothing I have to handle. 

Besides the client, I'm having trouble retrying to get the device twin when the connection comes back? It seems I can never successfully call **getTwin** after I've lost connection, getting a "_message timed out_" error. 

Today, I call **getTwin** after successfully opening the client. I close the client on the "disconnect" event and set a timeout for retrying to reconnect. When the connection comes online again I open the connection, but it always fail on trying to call **getTwin** the second time.

# Code sample exhibiting the issue:

```javascript
'use strict'

var token = "...";
var Client = require('azure-iot-device').Client;
var DeviceProtocol = require('azure-iot-device-mqtt').MqttWs; // Default transport for Receiver
var client = Client.fromSharedAccessSignature(token, DeviceProtocol);
var isDissconnected = true;

client.on('disconnect', function () {
    isDissconnected = true;
    console.log('Disconnected');
    client.close(function () {
        console.log('Closed');
    });
});
client.on('connected', function () {
    console.log('Connected');
});
client.on('error', function (err) {
    console.error('ERROR: ' + err.message);
});

var connectClient = function () {
    console.log('Connecting client');
    client.open(function (err, transport) {
        if (err) {
            console.log('Could not connect: ' + err);
        }
        else {
            isDissconnected = false;
            client.getTwin(function (err, twin) {
                if (err) {
                    console.log('Could not get twin: ' + err);
                }
                else {
                    console.log('Got twin');
                }
            });
        }
    });
}
connectClient();
setInterval(function () {
    if (isDissconnected) {
        console.log("Retrying to connect...");
        connectClient();
    }
}, 10000);
```
# Console log of the issue:
**>node app.js**
Connecting client
Got twin
Disconnected
Closed
Retrying to connect...
Connecting client
Could not get twin: TimeoutError: message timed out

**Comment by pierreca** [comment]
Thanks for the code, i've been able to repro the issue. looking into it!

**Comment by pierreca** [comment]
@wmmihaa I pushed a branch with a potential fix: https://github.com/Azure/azure-iot-sdk-node/tree/fix-28

that fix is in review with the rest of the team and we'll merge into master if the review is satisfactory, but I wanted to give you a headstart and see if it works for you.

As an aside, we have been working on reconnection logic and network tests, and we should make some progress in the next few weeks regarding that. The difficulty we have to deal with is that the logic has to function in the same way for HTTP, AMQP (which uses CBS authentication, meaning no need to disconnect on a regular basis) and MQTT (which needs to reconnect every time the SAS token must be renewed).

Thanks for the feedback and the code to help repro, it greatly helped fix this fast!

**Comment by wmmihaa** [comment]
Excellent @pierreca

**Comment by pierreca** [comment]
Fix has been checked into the master branch. Will be included in our next release (in the next few days). I'll close this issue when the packages are released.

**Comment by wmmihaa** [comment]
Well done @pierreca

**Comment by pierreca** [comment]
Published v1.1.9 with the bugfix. Thanks again!
